dask_ml/ensemble/_blockwise.py

```diff
@@ -0,0 +1,240 @@
+import dask
+import dask.array as da
+import dask.dataframe as dd
+import numpy as np
+import sklearn.base
+from sklearn.utils.validation import check_is_fitted
+
+from ..base import ClassifierMixin, RegressorMixin
+from ..utils import check_array
+
+
+class BlockwiseBase(sklearn.base.BaseEstimator):
+    def __init__(self, estimator):
+        self.estimator = estimator
+
+    def _check_array(self, X):
+        return check_array(
+            X,
+            accept_dask_dataframe=True,
+            accept_unknown_chunks=True,
+            preserve_pandas_dataframe=True,
+        )
+
+    def fit(self, X, y, **kwargs):
+        X = self._check_array(X)
+        estimatord = dask.delayed(self.estimator)
+
+        Xs = X.to_delayed()
+        ys = y.to_delayed()
+        if isinstance(X, da.Array):
+            Xs = Xs.flatten()
+        if isinstance(y, da.Array):
+            ys = ys.flatten()
+
+        estimators = [
+            dask.delayed(sklearn.base.clone)(estimatord) for _ in range(len(Xs))
+        ]
+        results = [
+            estimator_.fit(X_, y_, **kwargs)
+            for estimator_, X_, y_, in zip(estimators, Xs, ys)
+        ]
+        results = list(dask.compute(*results))
+        self.estimators_ = results
+
+    def _predict(self, X):
+        """Collect results from many predict calls"""
+        if isinstance(self, ClassifierMixin):
+            dtype = "int"
+        else:
+            dtype = "float"
+
+        if isinstance(X, da.Array):
+            results = [
+                X.map_blocks(_predict, dtype=dtype, estimator=estimator, drop_axis=1)
+                for estimator in self.estimators_
+            ]
+            combined = da.vstack(results).T.rechunk({1: -1})
```

Comment:
This is kinda slow for many estimators (many partitions). We end up with `X.npartitions * len(self.classifiers_)` tasks. Looking into doing a "batch" predict where each task represents the predictions from every estimator on that partition stacked into a single ndarray.

Reply:
Fixed for array. For dask.dataframe.map_partitions didn't especially like returning a 3D array. Looking into it a bit, but not a blocker right now.